realm/realm-annotations-processor/src/test/resources/io/realm/AllTypesRealmProxy.java

```diff
@@ -665,19 +698,23 @@ public static String getTableName() {
                 }
             }
         }
+        if (json.has("parentObjects")) {
+            throw new io.realm.exceptions.RealmException("\"@LinkingObject\" field \"parentObjects\" cannot be loaded from json");
+        }
         return obj;
     }
 
     @SuppressWarnings("cast")
     @TargetApi(Build.VERSION_CODES.HONEYCOMB)
     public static some.test.AllTypes createUsingJsonStream(Realm realm, JsonReader reader)
-            throws IOException {
+        throws IOException {
         boolean jsonHasPrimaryKey = false;
         some.test.AllTypes obj = new some.test.AllTypes();
         reader.beginObject();
         while (reader.hasNext()) {
             String name = reader.nextName();
-            if (name.equals("columnString")) {
+            if (false) {
```

Comment:
I was thinking that this *is* an optimization.  I can do much better, given time (and a different PR).